stdlib/array.gr

```diff
@@ -946,16 +946,18 @@ provide let slice = (start, end=length(array), array) => {
  * Sorts an array in-place.
  *
  * Ordering is calculated using a comparator function which takes two array elements and must return 0 if both are equal, a positive number if the first is greater, and a negative number if the first is smaller.
- * @param comp: The comparator function used to indicate sort order
  * @param array: The array to be sorted
+ * @param compare: The comparator function used to indicate sort order
+ * 
  * @since v0.4.5
+ * @history v0.6.0: Made `compare` a default argument
  */
-provide let sort = (comp, array) => {
+provide let sort = (array, compare=compare) => {
```

Comment:
Why did you change the argument order?

Reply:
No reason in particular other that I prefer default arguments appearing after required ones in the definition. I can change it back though

Reply:
I don't know if it's the OCamler in me or what, but I prefer the default arguments before the required ones. It's probably the OCamler in me.

Reply:
I love them at the end because it sorts them to the bottom of the documentation. So yeah, I think it's the ocamler in you 😂

Reply:
Additionally, default arguments at the end are familiar to JavaScript/TypeScript folks.

Reply:
I think as long as we're doing data-last in the rest of the stdlib then it makes sense to have the `compare` arg before the array/list. This is kinda the same issue/argument as https://github.com/grain-lang/grain/issues/671